Signal/src/ViewControllers/ConversationView/ConversationInputTextView.m

```diff
@@ -44,13 +52,89 @@ - (instancetype)init
 
         self.text = nil;
 
-        //        _placeHolder = nil;
-        //        _placeHolderTextColor = [UIColor lightGrayColor];
+        self.placeholderView = [UILabel new];
```

Comment:
There's two ways to add a placeholder to `UITextArea`, using a subview (as I do here) or by using `drawRect:`.  Overriding `drawRect:` has serious perf implementations so I've gone this way.

Reply:
👍